mm2src/coins/qrc20.rs

```diff
@@ -886,10 +891,15 @@ impl SwapOps for Qrc20Coin {
         Box::new(fut.boxed().compat())
     }
 
-    fn validate_maker_payment(&self, input: ValidatePaymentInput) -> Box<dyn Future<Item = (), Error = String> + Send> {
-        let payment_tx: UtxoTx = try_fus!(deserialize(input.payment_tx.as_slice()).map_err(|e| ERRL!("{:?}", e)));
-        let sender = try_fus!(self.contract_address_from_raw_pubkey(&input.other_pub));
-        let swap_contract_address = try_fus!(input.swap_contract_address.try_to_address());
+    fn validate_maker_payment(&self, input: ValidatePaymentInput) -> ValidatePaymentFut<()> {
+        let payment_tx: UtxoTx = try_f!(deserialize(input.payment_tx.as_slice()));
+        let sender = try_f!(self
+            .contract_address_from_raw_pubkey(&input.other_pub)
+            .map_to_mm(ValidatePaymentError::InvalidInput));
+        let swap_contract_address = try_f!(input
+            .swap_contract_address
+            .try_to_address()
+            .map_to_mm(ValidatePaymentError::AddressParseError));
```

Comment:
This comment is still not solved https://github.com/KomodoPlatform/atomicDEX-API/pull/1444#discussion_r946509784

Reply:
done